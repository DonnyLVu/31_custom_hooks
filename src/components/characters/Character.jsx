import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, photoUrl }) => (
  <figure data-testid="character">
    <figcaption>{name}</figcaption>
    <img src={photoUrl} />
  </figure>
);

Character.propTypes = {
  _id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  photoUrl: PropTypes.string
};

export default Character;
