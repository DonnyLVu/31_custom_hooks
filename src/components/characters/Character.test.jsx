import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Character component', () => {
  afterEach(() => cleanup());
  it('renders Character', () => {
    const { asFragment } = render(<Router><Character /></Router>);
    expect(asFragment()).toMatchSnapshot();
  });
});
