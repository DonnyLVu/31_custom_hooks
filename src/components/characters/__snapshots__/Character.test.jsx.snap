// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <figure
    data-testid="character"
  >
    <figcaption />
    <img />
  </figure>
</DocumentFragment>
`;
